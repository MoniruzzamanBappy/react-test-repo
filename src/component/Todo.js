import React from 'react';

const Todo = ({data}) => {
    const { id, title, completed } = data;
    const h1 = <h1>{title}</h1>;
    const p = completed ? <strike>{h1}</strike> : h1;
    return (
        <div data-testid={`todo-${id}`}>{p}</div>
    );
};

export default Todo;