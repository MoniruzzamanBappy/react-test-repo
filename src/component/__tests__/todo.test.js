import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../Todo";
afterEach(() => {
  cleanup();
});
test("should render  todo 1 compnent", () => {
  const datas = {
    id: 1,
    title: "Hello world",
    completed: false,
  };
  render(<Todo data={datas} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Hello world");
});
test("should render todo 2 compnent", () => {
  const datas = {
    id: 2,
    title: "Hello world 2",
    completed: false,
  };
  render(<Todo data={datas} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Hello world 2");
  expect(todoElement).not.toContainHTML("<strike>");
});
test("should render todo 3 compnent", () => {
  const datas = {
    id: 3,
    title: "Hello world 3",
    completed: true,
  };
  render(<Todo data={datas} />);
  const todoElement = screen.getByTestId("todo-3");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Hello world 3");
  expect(todoElement).toContainHTML("<strike>");
});

test("take a snap shot", () => {
  const datas = {
    id: 1,
    title: "Hello world",
    completed: false,
  };
  const { asFragment } = render(<Todo data={datas} />);
  expect(asFragment()).toMatchSnapshot();
});
