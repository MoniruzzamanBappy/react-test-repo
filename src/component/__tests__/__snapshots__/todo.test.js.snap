// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a snap shot 1`] = `
<DocumentFragment>
  <div
    data-testid="todo-1"
  >
    <h1>
      Hello world
    </h1>
  </div>
</DocumentFragment>
`;
